### PR TITLE
libbpf-tools: convert BCC tcpsynbl to BPF CO-RE version

### PR DESCRIPTION
**Pull request by chendotjs**
Signed-off-by: chendotjs <chendotjs@gmail.com>

### CODE REVIEW

File: libbpf-tools/tcpsynbl.c [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	if (env.ipv4) {
+		set_autoload_prog(obj, 4, true);
+	} else if (env.ipv6) {
+		set_autoload_prog(obj, 6, true);
+	} else {
+		set_autoload_prog(obj, 4, false);
+		set_autoload_prog(obj, 6, false);
+	}
+
+  err = tcpsynbl_bpf__load(obj);
```
Comment (chenhengqi) [inline libbpf-tools/tcpsynbl.c:217]: wrong indention.

File: libbpf-tools/tcpsynbl.c [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	if (err)
+		return err;
+
+	libbpf_set_print(libbpf_print_fn);
+
+	err = bump_memlock_rlimit();
+	if (err) {
+		fprintf(stderr, "failed to increase rlimit: %d\n", err);
+		return 1;
+	}
```
Comment (chenhengqi) [inline libbpf-tools/tcpsynbl.c:200]: Replace these with `libbpf_set_strict_mode(LIBBPF_STRICT_ALL);`.

File: libbpf-tools/tcpsynbl.c (hunk 1 above)
Comment (chendotjs) [reply libbpf-tools/tcpsynbl.c]: indention fixed, thanks

File: libbpf-tools/tcpsynbl.c (hunk 2 above)
Comment (chendotjs) [reply libbpf-tools/tcpsynbl.c]: done

**Comment by chendotjs** [comment]
@anakryiko @chenhengqi PTAL when you have time, thank you  :)

File: libbpf-tools/tcpsynbl.c [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+		pos_args++;
+		break;
+	default:
+		return ARGP_ERR_UNKNOWN;
+	}
+	return 0;
+}
+
+int libbpf_print_fn(enum libbpf_print_level level,
+		    const char *format, va_list args)
```
Comment (chenhengqi) [inline libbpf-tools/tcpsynbl.c:105]: static, and please keep this signature on a single line.

Comment (chendotjs) [reply libbpf-tools/tcpsynbl.c]: @chenhengqi resolved

File: libbpf-tools/tcpsynbl.c [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+	printf("Tracing SYN backlog size. Ctrl-C to end.\n");
+
+	/* main: poll */
+	while (1) {
+		sleep(env.interval);
+		printf("\n");
+
+		if (env.timestamp) {
+			time(&t);;
```
Comment (chenhengqi) [inline libbpf-tools/tcpsynbl.c:235]: double `;` ? 

Can you move all these print into `print_log2_hists` ?

File: libbpf-tools/tcpsynbl.c [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+		return 0;
+	return vfprintf(stderr, format, args);
+}
+
+static void sig_handler(int sig)
+{
+	exiting = true;
+}
+
+static void set_autoload_prog(struct tcpsynbl_bpf *obj, int version, bool exclusive)
```
Comment (chenhengqi) [inline libbpf-tools/tcpsynbl.c:116]: Can this be simplified ?https://github.com/iovisor/bcc/blob/d3475ad142e080615e25ad3ad192cdf097af8726/tools/tcpsynbl.py#L62-L68

**Comment by davemarchevsky** [comment]
Sorry for the delay here. I think this is good to go - ran it a few times, compared output to python tool.

**Pull request merged**